util/codec/collation_test.go

```diff
@@ -120,15 +121,15 @@ func (s *testCollationSuite) TestHashChunkRowCollation(c *C) {
 	for i := 0; i < n; i++ {
 		h1 := crc32.NewIEEE()
 		h2 := crc32.NewIEEE()
-		c.Assert(HashChunkRow(sc, h1, chk1.GetRow(i), tps, cols, buf), IsNil)
-		c.Assert(HashChunkRow(sc, h2, chk2.GetRow(i), tps, cols, buf), IsNil)
-		c.Assert(h1.Sum32(), Equals, h2.Sum32())
+		require.NoError(t, HashChunkRow(sc, h1, chk1.GetRow(i), tps, cols, buf))
+		require.NoError(t, HashChunkRow(sc, h2, chk2.GetRow(i), tps, cols, buf))
+		require.Equal(t, h2.Sum32(), h1.Sum32())
 		h1.Reset()
 		h2.Reset()
 	}
 }
 
-func (s *testCollationSuite) TestHashChunkColumnsCollation(c *C) {
+func TestHashChunkColumnsCollation(t *testing.T) {
```

Comment:
seems can't be parallel, because this test used `collate.SetNewCollationEnabledForTest` method which will set a global variable `newCollationEnabled,` and this method also be invoked in `TestHashChunkColumnsCollation`

Reply:
I'd suggest factor out these two tests in a new test file. We don't need a different test binary but mix parallel and non-parallel is brittle.

Reply:
thanks for your feedback. 
in fact, `collation_test.go` all three tests used `collate.SetNewCollationEnabledForTest` method, so may i remove `t.Parallel()` in first test and make `collation_test.go` run in serial ? 

but this test package binary still contains parallel and non-parallel tests, any suggestion ? thanks a lot

Reply:
> in fact, collation_test.go all three tests used collate.SetNewCollationEnabledForTest method, so may i remove t.Parallel() in first test and make collation_test.go run in serial ?

This makes sense.

> but this test package binary still contains parallel and non-parallel tests, any suggestion ? thanks a lot

It is ok if you think a bit more how parallel and non-parallel tests run together. I suggest separate tests in different test "files" in order not to mislead developer, but the compiler will handle it properly.

Reply:
okay